plugin/builtin/taskdata/ui.go

```diff
@@ -112,6 +112,28 @@ func uiHandleTaskTag(w http.ResponseWriter, r *http.Request) {
 	util.WriteJSON(w, http.StatusOK, "")
 }
 
+// Wrapper `GetDistinctTagNames` model function
+// requires HTTP GET param `project_id`
+// Return distinct list of tags for given `project_id`
+func uiGetProjectTags(w http.ResponseWriter, r *http.Request) {
+	vals, ok := r.URL.Query()["project_id"]
```

Comment:
Could you get this from the project context rather than a query string?

Reply:
It mostly depend on do we trying to build REST-like api or not.